python/lib/dicom/summary_make.py

```diff
@@ -0,0 +1,218 @@
+from functools import cmp_to_key
+import os
+import pydicom
+import pydicom.errors
+from lib.dicom.summary_type import Summary, Info, Patient, Scanner, Acquisition, DicomFile, OtherFile
+from lib.dicom.text import make_hash, read_dicom_date_none
+
+
+def get_value(dicom: pydicom.Dataset, tag: str):
+    """
+    Get a required value from a DICOM.
+    """
+    if tag not in dicom:
+        raise Exception(f'Expected DICOM tag \'{tag}\' but found none.')
+
+    return dicom[tag].value
+
+
+def get_value_none(dicom: pydicom.Dataset, tag: str):
+    """
+    Get a nullable value from a DICOM.
+    """
+    if tag not in dicom:
+        return None
+
+    return dicom[tag].value or None
+
+
+def cmp_int_none(a: int | None, b: int | None):
+    """
+    Order comparison between two nullable integers.
+    """
+    match a, b:
+        case None, None:
+            return 0
+        case _, None:
+            return -1
+        case None, _:
+            return 1
```

Comment:
`_` is the wildcard pattern, which kind of means "anything works" (so any `Some(x)` such as `Some(0)`, `Some(1)`... work here, since `None` was already matched before).
The returns `-1`, `1`... are because this is a comparison function, a negative number means that `a < b`, `0` that `a == b`, and a positive number that `a > b`. This is required by `cmp_to_key` in `summary_make` but I'll add some documentation.